### PR TITLE
Fix authority in uri_parser

### PR DESCRIPTION
**Pull request by hongweiwang**
Fix authority and path in uri_parser.c. Authority begins with "//" and ends with the next "/".

### CODE REVIEW

**Comment by hongweiwang** [comment]
I signed it!

File: src/core/client_config/uri_parser.c [hunk 1]
```diff
@@ -99,7 +99,7 @@ grpc_uri *grpc_uri_parse(const char *uri_text, int suppress_errors) {
   if (uri_text[scheme_end + 1] == '/' && uri_text[scheme_end + 2] == '/') {
     authority_begin = scheme_end + 3;
     for (i = authority_begin; uri_text[i] != 0; i++) {
-      if (uri_text[i] == '/') {
+      if (uri_text[i] == '/' && authority_end == -1) {
```
Comment (ctiller) [inline src/core/client_config/uri_parser.c]: Wouldn't it be better to put the `&& authority_end == -1` clause in the for loop, to make its termination condition `uri_text[i] != 0 && authority_end == -1`?

Comment (ctiller) [inline src/core/client_config/uri_parser.c]: (That way we avoid redundantly scanning the rest of the string, and possibly making other errors)

**Comment by hongweiwang** [comment]
Done

File: src/core/client_config/uri_parser.c (hunk 1 above)
Comment (hongweiwang) [inline src/core/client_config/uri_parser.c]: Done

**Pull request merged**